redux/slices/cctCalcSlice.ts

```diff
@@ -0,0 +1,50 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import dayjs from "dayjs";
+
+type CctCalcState = {
+  modalOpen: boolean;
+  progName: string;
+  currentProgEndDate: Date | string;
+  newEndDates: any[];
+  propStartDate: Date | string;
+};
+
+const initialState: CctCalcState = {
+  modalOpen: false,
+  progName: "",
+  currentProgEndDate: "",
+  newEndDates: [],
+  propStartDate: dayjs().add(16, "weeks").format("YYYY-MM-DD")
```

Comment:
Maybe this should default to the programme start date if this is more than 16 weeks in the future? (And not be allowed to be before the programme start date)